Tail subsystem and commands

Onkornix opened the pull request:
need guidance on the tail commands whether to use a relative or absolute encoder, or if my method is ok. also if I did the PID loops correctly.

mads841 committed:
created tail subsystem and crated tail motor object and init

mads841 committed:
finished tail subsystem I believe

Onkornix committed:
testing git config

Onkornix committed:
changed position conversion factor to a constant in tail subsystem

Onkornix committed:
Attempted tail up and tail down commands w/ PID

I explain my trouble in TailSubsystem.kt

fruzzmuffin commented on src/main/java/org/sert2521/bunnybots2024/commands/TailDownCommand.kt:
**TailUpCommand** and **TailDownCommand** can be one file with the setpoint as an initializing argument

fruzzmuffin commented on src/main/java/org/sert2521/bunnybots2024/subsystems/TailSubsystem.kt:
```diff
@@ -0,0 +1,48 @@
+    same angles every time. I imagine that can lead to a compounding of slight
+    errors over time.
+     */
+    private const val POSITION_CONVERSION_FACTOR = Math.PI/24
```
Constants don't go in subsystem files. To be honest I would just get rid of it and just set the position conversion factor straight to the value instead of having a constant in between

fruzzmuffin commented on src/main/java/org/sert2521/bunnybots2024/subsystems/TailSubsystem.kt:
Since we are using a relative encoder instead of an absolute encoder, we should probably have a function that resets the encoder to a certain value. Since we don't have a feedforward here the literal value isn't needed, so I would just set it to zero.

fruzzmuffin commented on src/main/java/org/sert2521/bunnybots2024/commands/TailDownCommand.kt:
```diff
@@ -0,0 +1,41 @@
+    }
+
+    override fun isFinished(): Boolean {
+        return tailPid.atSetpoint()
```
I don't think we should end this command. If the command ends after it hits the setpoint, then after it hits its setpoint it will go limp (with braking mode but still). I don't believe braking mode is enough to pull totes on its own. You can just make this return false and let the next command to the tail cancel this one through the default behavior of subsystem requirements.

fruzzmuffin replied:
it's either that or make the **defaultCommand** of Tail be a version of this command with a goal that keeps it at its current location

Onkornix commented:
Alright I'll try to be there tomorrow and implement this stuff. Thanks!

Onkornix committed:
Implemented suggested changes

Onkornix commented:
- I changed it to one command that takes the setpoint as an argument
- i removed the constant in the subsystem
- made a default command for TailSubsystem that holds the motor still (I think)
- made the encoder's position be set to pi/2 if the tail was in the lowered position when the motor is shut off
